### PR TITLE
Add parallel tests execution to CI

### PR DESCRIPTION
**Pull request by smiasojed**
## Summary
Closes #_
- [ ] y/n | Does it introduce breaking changes?
- [ ] y/n | Is it dependant on the specific version of `cargo-contract` or `pallet-contracts`?

## Description
Add `nextest` to the workflow where parallel test execution makes sense.
In cases where parallel execution of tests cannot be used, split the tests between different machines.

## Checklist before requesting a review
- [ ] My code follows the style guidelines of this project
- [ ] I have added an entry to `CHANGELOG.md`
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by smiasojed** [comment]
Time for `test` job decreased by 50%, from 18min to 9 min

**Comment by smiasojed** [comment]
Each end-to-end test takes between 50 to 90 seconds. Currently, they are executed iteratively because end-to-end tests build contracts during test execution, and the same contracts are rebuilt by multiple tests instead of reusing them, causing a race condition.

**Comment by smiasojed** [comment]
The `examples-test` job is split into 4 parallel jobs: linear duration decreased from 36 to 12 minutes, but when we sum up the 4 parallel jobs, we get 48 minutes (the cost of launching and compilation of common crates on each machine)

**Comment by smiasojed** [comment]
Total time has change from 70 to 50 minutes

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -444,7 +444,7 @@ jobs:
           git config --global --add safe.directory '*'
           # RUSTFLAGS are the cause target cache can't be used here
           cargo build --all-features --workspace
-          cargo test --all-features --no-fail-fast --workspace --locked
+          cargo test --all-features --no-fail-fast --workspace --locked -- --skip ui_tests
```
Comment (cmichi) [inline .github/workflows/ci.yml:447]: Why the `--skip ui_tests` here?

Comment (cmichi) [inline .github/workflows/ci.yml:447]: Good idea to skip it for `codecov`!

**Review by cmichi — APPROVED** [review]
This is some good work! Thanks, Sebastian!

**Review by ascjones — APPROVED** [review]
:rocket:

**Pull request merged**